py/server/deephaven/table.py

```diff
@@ -1324,6 +1325,31 @@ def partition_by(self, by: Union[str, Sequence[str]], drop_keys: bool = False) -
         except Exception as e:
             raise DHError(e, "failed to create a partitioned table.") from e
 
+    def update_by(self, clauses: Union[UpdateByClause, List[UpdateByClause]],
+                  by: Union[str, List[str]] = None) -> Table:
+        """This method applies the specified row based operations to this table on a cumulative basis within the
+        groups and creates a new table that has the same set of row as this table. As opposed to
+        :meth:`~Table.update` which is strictly individual row based, these operations are capable of processing
+        state between rows within the same row groups which are identified by the provided key columns. If no key
+        columns are provided, all the rows of the table forms a single group.
+
+        Args:
+            clauses (Union[UpdateByClause, List[UpdateByClause]]): the update-by operation definition(s)
```

Comment:
I agree that the term 'operation' is better and I think we better change 'updateByClause' to 'updateByOperation'.  Since we don't diverge too far from the Java code, which we have been able to do so far, it is best that we do the renaming in Java as well. Thoughts?

Reply:
Ok with me.

Reply:
Created PR #2680 that renamed this object to `UpdateByOperations`